src/wait-for-dom-change.js

```diff
@@ -0,0 +1,37 @@
+import 'mutationobserver-shim'
+
+function waitForDomChange({
+  container = document,
+  timeout = 4500,
+  mutationObserverOptions = {
+    subtree: true,
+    childList: true,
+    attributes: true,
+    characterData: true,
+  },
+} = {}) {
+  return new Promise((resolve, reject) => {
+    // Disabling eslint prefer-const below: either prefer-const or no-use-before-define triggers.
+    let lastError, observer, timer // eslint-disable-line prefer-const
+    function onDone(error, result) {
+      clearTimeout(timer)
+      setImmediate(() => observer.disconnect())
+      if (error) {
+        reject(error)
+      } else {
+        resolve(result)
+      }
+    }
+    function onMutation() {
+      onDone(null, true)
+    }
+    function onTimeout() {
+      onDone(lastError || new Error('Timed out in waitForDomChange.'), null)
+    }
+    timer = setTimeout(onTimeout, timeout)
+    observer = new window.MutationObserver(onMutation)
+    observer.observe(container, mutationObserverOptions)
```

Comment:
Should this MutationObserver code be reused somehow, not copied, between waitForElement and waitForDomChange?

Reply:
I'd prefer if we could reuse the code 👍

Reply:
How do you propose we reuse the code?